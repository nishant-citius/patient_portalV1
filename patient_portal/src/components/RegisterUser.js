import React from "react";

const RegisterUser = () => {
  return (
    <>
      
      <div className="container">
      <h4 className="text-center">Register Page</h4>
        <div className="row justify-content-center">
          <div className="col-8">
          <form name="registration_form" method="Post">
            <div className="form-group">
              <label>First Name</label>
              <input type="text" className="form-control" name="first_name" id="first_name" placeholder="Enter Your First name"/>
            </div>
            <div className="form-group">
              <label>Last Name</label>
              <input type="text" className="form-control" name="last_name" id="last_name" placeholder="Enter Your Last name"/>
            </div>
            <div className="form-group">
              <label>DOB</label>
              <input type="text" className="form-control" name="dob" id="dob" placeholder="Enter Your Dob"/>
            </div>
            <div className="form-group">
              <label>User Name</label>
              <input type="text" className="form-control" name="user_name" id="user_name" placeholder="Enter User Name"/>
            </div>
            <div className="form-group">
              <label>Role</label>
              <select className="form-control" name="role" id="role">
                <option value="">Select</option>
                <option value="admin">Admin</option>
                <option value="patient">Patient</option>
                <option value="physician">Physician</option>
              </select>
            </div>
            <div className="form-group">
              <label>Email</label>
              <input type="email" className="form-control" id="email" name="email"  placeholder="Enter email"/>
            </div>
            <div className="form-group">
              <label>Password</label>
              <input type="password" className="form-control" id="password" placeholder="Password"/>
            </div>
            <div className="form-group">
              <label>Retype-Password</label>
              <input type="password" className="form-control" id="re_password" placeholder="Password"/>
            </div>
            <button type="submit" className="btn btn-primary">Submit</button>
          </form>
          </div>
        </div>
      </div>
    </>
  );
};

export default RegisterUser;
