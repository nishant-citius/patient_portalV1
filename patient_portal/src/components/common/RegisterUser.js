import { React, useState, useEffect } from "react";
import { connect } from "react-redux";
import * as actionCreator from "../../redux/actions/userActionCreater";
import { useHistory } from "react-router";
import { Formik, Form, Field, ErrorMessage } from "formik";
import * as Yup from "yup";
import { Grid, Paper, Avatar, TextField } from "@material-ui/core";
import LockOpenIcon from "@material-ui/icons/LockOpen";

const RegisterUser = (props) => {
  let history = useHistory();

  const initialValues = {
    fName: "",
    lName: "",
    dob: "",
    username: "",
    email: "",
    mobile: "",
    role: "patient",
    password: "",
    rpassword: "",
    blood_group: "",
    createdDate: Date(),
    isActive: false,
  };

  const validationSchema = Yup.object().shape({
    fName: Yup.string().required("Required"),
    lName: Yup.string().required("Required"),
    dob: Yup.string().required("Required"),
    username: Yup.string().required("Required"),
    email: Yup.string().email("Please enter valid email").required("Required"),
    password: Yup.string()
      .required("This field is required")
      .min(4, "Password should be of minimum 4 characters length"),
    rpassword: Yup.string().when("password", {
      is: (val) => (val && val.length > 0 ? true : false),
      then: Yup.string().oneOf(
        [Yup.ref("password")],
        "Both password need to be the same"
      ),
    }),
    mobile: Yup.string()
      .required("Required")
      .min(10, "Phone number required 10 digit")
      .max(12, "Phone number required 12 digit"),
    blood_group: Yup.string().required("Required"),
  });

  const onSubmit = (values) => {
    const payload = {
      fName: values.fName,
      lName: values.lName,
      dob: values.dob,
      username: values.username,
      email: values.email,
      education: values.education,
      password: values.password,
      rpassword: values.rpassword,
      mobile: values.mobile,
      blood_group: values.blood_group,
      role: "patient",
      createdDate: Date(),
      isActive: false,
    };
    props.register(payload);
  };

  useEffect(() => {
    if (props.statusCode === 200) {
      props.flashNotification({
        message: "Registration Succeessful...",
        type: "success",
      });
      history.push("/login");
    }
  });

  const paperStyle = {
    padding: 20,
    height: "",
    width: 400,
    margin: "60px auto",
    marginTop: "110px",
  };
  const avatarStyle = { backgroundColor: "#1bbd7e" };
  const btnstyle = { margin: "8px 0" };
  return (
    <>
      <Paper elevation={10} style={paperStyle}>
        <Grid align="center">
          <Avatar style={avatarStyle}>
            <LockOpenIcon />
          </Avatar>
          <br />
          <h4>Registration</h4>
          <br />
        </Grid>
        <Formik
          initialValues={initialValues}
          onSubmit={onSubmit}
          validationSchema={validationSchema}
        >
          {(props) => (
            <Form>
              <Field
                as={TextField}
                margin="normal"
                label="First Name"
                placeholder="enter first name"
                fullWidth
                name="fName"
                id="fName"
                variant="filled"
                helperText={<ErrorMessage name="fName" />}
              />
              <Field
                as={TextField}
                margin="normal"
                label="Last Name"
                placeholder="enter last name"
                fullWidth
                name="lName"
                id="lName"
                variant="filled"
                helperText={<ErrorMessage name="lName" />}
              />
              <Field
                as={TextField}
                margin="normal"
                label="D.O.B"
                type="date"
                className="pt_10"
                fullWidth
                name="dob"
                id="dob"
                variant="filled"
                helperText={<ErrorMessage name="dob" />}
              />
              <Field
                as={TextField}
                margin="normal"
                type="text"
                label="UserName"
                placeholder="enter username"
                fullWidth
                name="username"
                id="username"
                variant="filled"
                helperText={<ErrorMessage name="username" />}
              />
              <Field
                as={TextField}
                margin="normal"
                label="Role"
                fullWidth
                name="role"
                id="role"
                defaultValue="Patient"
                variant="filled"
                disabled
              />
              <Field
                as={TextField}
                margin="normal"
                type="number"
                label="Mobile"
                placeholder="enter mobile no"
                fullWidth
                name="mobile"
                id="mobile"
                variant="filled"
                helperText={<ErrorMessage name="mobile" />}
              />
              <Field
                as={TextField}
                margin="normal"
                placeholder="Enter Blood Group"
                fullWidth
                name="blood_group"
                id="blood_group"
                label="Blood Group"
                variant="filled"
                helperText={<ErrorMessage name="blood_group" />}
              />
              <Field
                as={TextField}
                margin="normal"
                type="text"
                label="Email"
                placeholder="enter email"
                fullWidth
                name="email"
                id="email"
                variant="filled"
                helperText={<ErrorMessage name="email" />}
              />
              <Field
                as={TextField}
                margin="normal"
                type="password"
                label="Password"
                placeholder="enter password"
                fullWidth
                name="password"
                id="password"
                variant="filled"
                helperText={<ErrorMessage name="password" />}
              />
              <Field
                as={TextField}
                margin="normal"
                type="password"
                label="Confirm password"
                placeholder="enter rpassword"
                fullWidth
                name="rpassword"
                id="rpassword"
                variant="filled"
                helperText={<ErrorMessage name="rpassword" />}
              />
              <div>
                <button type="submit" className="btn btn-primary mt-2 center">
                  Submit
                </button>
              </div>
            </Form>
          )}
        </Formik>
        <h4 className="text-danger">{props.globalMessage}</h4>
      </Paper>
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    globalMessage: state.register.globalmessage,
    statusCode: state.register.statusCode,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    register: (newuser) => dispatch(actionCreator.Register(newuser)),
  };
};

let hof = connect(mapStateToProps, mapDispatchToProps);
export default hof(RegisterUser);
