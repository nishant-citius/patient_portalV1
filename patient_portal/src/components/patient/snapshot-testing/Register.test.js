import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import RegisterUser from "../../common/RegisterUser";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

describe("Register snapshot test", () => {
  let store;
  let jsx;

  beforeEach(() => {
    store = mockStore({ register: {} });
    jsx = (
      <Provider store={store}>
        <RegisterUser />
      </Provider>
    );
  });
  it("should render properly", () => {
    const tree = renderer.create(jsx).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
