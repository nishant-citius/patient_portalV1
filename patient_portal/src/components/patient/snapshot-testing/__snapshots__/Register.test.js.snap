// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register snapshot test should render properly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
  style={
    Object {
      "height": "",
      "margin": "60px auto",
      "marginTop": "110px",
      "padding": 20,
      "width": 400,
    }
  }
>
  <div
    align="center"
    className="MuiGrid-root"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
      style={
        Object {
          "backgroundColor": "#1bbd7e",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h1.9c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10z"
        />
      </svg>
    </div>
    <br />
    <h4>
      Registration
    </h4>
    <br />
  </div>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="fName"
        id="fName-label"
      >
        First Name
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="fName-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="fName"
          name="fName"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter first name"
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="fName-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="lName"
        id="lName-label"
      >
        Last Name
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="lName-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="lName"
          name="lName"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter last name"
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="lName-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="dob"
        id="dob-label"
      >
        D.O.B
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="dob-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="dob"
          name="dob"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="dob-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="username"
        id="username-label"
      >
        UserName
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="username-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="username"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter username"
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="username-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-disabled Mui-disabled MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="role"
        id="role-label"
      >
        Role
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
          defaultValue="Patient"
          disabled={true}
          id="role"
          name="role"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value="patient"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="mobile"
        id="mobile-label"
      >
        Mobile
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="mobile-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="mobile"
          name="mobile"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter mobile no"
          required={false}
          type="number"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="mobile-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="blood_group"
        id="blood_group-label"
      >
        Blood Group
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="blood_group-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="blood_group"
          name="blood_group"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter Blood Group"
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="blood_group-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        Email
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="email-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="email"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter email"
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="email-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Password
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="password-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="password"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter password"
          required={false}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="password-helper-text"
      />
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink={false}
        htmlFor="rpassword"
        id="rpassword-label"
      >
        Confirm password
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="rpassword-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          id="rpassword"
          name="rpassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter rpassword"
          required={false}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="rpassword-helper-text"
      />
    </div>
    <div>
      <button
        className="btn btn-primary mt-2 center"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
  <h4
    className="text-danger"
  />
</div>
`;
